pynvim/api/nvim.py

```diff
@@ -76,7 +76,7 @@ def from_session(cls, session):
         queries Nvim metadata for type information and sets a SessionHook for
         creating specialized objects from Nvim remote handles.
         """
-        session.error_wrapper = lambda e: NvimError(e[1])
+        session.error_wrapper = lambda e: NvimError(e[1].decode())
```

Comment:
This is done to have strings with exception arguments, instead of bytes (py3).

Reply:
Should I take this out of here for now already?

Or can we finish it via this PR?

Reply:
Should use `decode("utf-8", "surrogateescape")` on python3 and no decode on python2 ( might be some function to reuse that does this)

Reply:
> no decode on python2

Why not?  Do we explicitly want no unicode string?

Reply:
There's `decode_if_bytes`.  Taken out into https://github.com/neovim/pynvim/pull/412.

Reply:
We are explicitly using the standard `str` type in both versions per default. Trying to "improve" string semantics on py2 is a non-goal at this point.